.github/workflows/format.yml

```diff
@@ -0,0 +1,35 @@
+name: format code
+on:
+  pull_request:
+
+jobs:
+  format-code:
+    name: Format Code
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package: [flutter, dart]
+    defaults:
+      run:
+        working-directory: ${{ matrix.package }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dart-lang/setup-dart@6a218f2413a3e78e9087f638a238f6b40893203d # pin@v1
+        if: ${{ matrix.package == 'dart' }}
+      - uses: subosito/flutter-action@48cafc24713cca54bbe03cdc3a423187d413aafa # pin@v2.10.0
+        if: ${{ matrix.package == 'flutter' }}
+
+      - name: pub get
+        run: ${{ matrix.package }} pub get
+
+      - name: format
+        run: dart format .
```

Comment:
Do we actually need to call `format`? doesn't `fix --apply` does it internally?

Reply:
No, `fix --apply` handles issues uncovered by `dart analyze`, not the once by `dart format`